### PR TITLE
RDM-3859 Add case view feature tests. Introduce a way to populate fie…

### PR DESCRIPTION
**Pull request by andrewfolga**
…ld by id with provided value for fixed list (dropdown), text field, yes no field (radio). Introduce localizing text field labels by id. Introduce mandatory fields for case type in TestData.js.

### Change description ###
Add functional test coverage.
Introduce a way to populate field by id with provided value for fixed list (dropdown), text field, yes no field (radio).
Introduce localizing text field labels by id.
Introduce mandatory fields for case type in TestData.js.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: test/stepDefinitions/caseDetailsSteps.js [hunk 1]
```diff
@@ -32,22 +32,43 @@ defineSupportCode(function ({ Given, When, Then, Before, After }) {
...
+  Then('the success case created bar will be visible', async function() {
+      let alertBarText = await caseDetailsPage.getSuccessAlertBarText();
+      expect(alertBarText).to.match(/^Case #\d{4}-\d{4}-\d{4}-\d{4} has been created\.$/);
+  });
+
   Then(/^the '(.*)' field will be visible on the '(.*)' tab$/, async function (tabfield, tabName) {
       await caseDetailsPage.clickTab(tabName);
-      let fields =await caseDetailsPage.getTabFields()
-      expect(fields).to.include(tabfield)
+      let fields =await caseDetailsPage.getTabFields();
```
Comment (tzarsmango) [inline test/stepDefinitions/caseDetailsSteps.js]: small typo `let fields = await...`

File: test/features/caseView.feature [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  Background:
+    Given I have logged in
+    And a case containing Tabs functionality exists
+    And I create the case
+
+Scenario: case reference visible on case view
+   Then the case reference will be visible and formatted well
+
+  Scenario: tabs visible on case view
```
Comment (andrewfolga) [inline test/features/caseView.feature]: Remove spacing

File: test/features/caseView.feature [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  Scenario: tab fields visible on case view
+    When I navigate to tab 'Tab2'
+    Then the following fields will be visible:
+      | Text Field 2        |
+      | Text Field 3        |
+      | Text Field 4        |
+
+  Scenario: print button visible on case view
+   Then the print button will be visible
```
Comment (andrewfolga) [inline test/features/caseView.feature]: Add newline

File: test/pageObjects/caseDetailsPage.js [hunk 4]
```diff
@@ -33,6 +36,45 @@ class CaseDetailsPage extends BasePage {
...
+   * @returns {Promise<boolean>}
+   */
+
+  async isPrintButtonReady() {
+    return await $(this._printButton).isDisplayed()
+        && await $(this._printButton).isEnabled();
+  }
+
+  /**
+   * Get text value for the latest event in the History tab
```
Comment (andrewfolga) [inline test/pageObjects/caseDetailsPage.js]: Fix the comment

File: test/stepDefinitions/baseSteps.js [hunk 5]
```diff
@@ -15,7 +15,9 @@ module.exports = {
     // This will create a case for 'All Field Data Types' caseType or any other case
     //that has a optional text field and no other mandatory fields
     let wizardPage = new CreateCaseWizardPage();
-    await wizardPage.interactWithField('text', 'text');
+    for (const elem of TestData.mandatoryFields) {
```
Comment (andrewfolga) [inline test/stepDefinitions/baseSteps.js]: Make sure to preserver default logic.

File: test/stepDefinitions/baseSteps.js [hunk 6]
```diff
@@ -15,7 +15,9 @@ module.exports = {
     // This will create a case for 'All Field Data Types' caseType or any other case
```
Comment (andrewfolga) [inline test/stepDefinitions/baseSteps.js]: Adjust the comment.

File: test/utils/fieldUtils.js [hunk 7]
```diff
@@ -93,24 +93,24 @@ class FieldDataTypes {
   }
 
   /**
-   * Select a random option from the dropdown
+   * Select a provided option from the dropdown
```
Comment (andrewfolga) [inline test/utils/fieldUtils.js]: or random

File: test/utils/fieldUtils.js [hunk 8]
```diff
@@ -93,24 +93,24 @@ class FieldDataTypes {
...
-    let fixedListField = await new CCDFixedListField(css);
-    await fixedListField.selectOption();
+    let fixedListField = await new CCDFixedListField(css, id);
+    await fixedListField.selectOption(value);
     return fixedListField;
   }
 
   /**
-   * Select random radio butto option
+   * Select a provided radio butto option
```
Comment (andrewfolga) [inline test/utils/fieldUtils.js]: or random

File: test/features/caseView.feature (hunk 2 above)
Comment (andrewfolga) [reply test/features/caseView.feature]: Done

File: test/utils/fieldUtils.js (hunk 7 above)
Comment (noronhaa) [reply test/utils/fieldUtils.js]: perhaps we should explicitly state in the doc here that if the value parameter is left blank then a random option will be chosen

File: test/utils/fieldUtils.js (hunk 8 above)
Comment (noronhaa) [reply test/utils/fieldUtils.js]: perhaps we should explicitly state in the doc here that if the value parameter is left blank then a random option will be chosen

File: test/stepDefinitions/baseSteps.js [hunk 9]
```diff
@@ -11,13 +11,29 @@ module.exports = {
...
-    //that has a optional text field and no other mandatory fields
     let wizardPage = new CreateCaseWizardPage();
-    await wizardPage.interactWithField('text', 'text');
+    await this.fillOutMandatoryFields();
+    await this.fillOutOptionalFields();
     await wizardPage.clickContinueButton();
     await wizardPage.clickSubmitCaseButton();
+  },
+
+  fillOutMandatoryFields: async function(){
```
Comment (noronhaa) [inline test/stepDefinitions/baseSteps.js]: nice!

**Pull request merged**